Can't run Python script from anywhere, even after PATH variable was edited? <body> I was trying to set up my Mac to run custom Python scripts from anywhere without having to type codepython [script.py]/code or the explicit file path.   The script is called "testscript". I put the script in my own bin directory (separate from code/usr/local/bin/code), called code/User/Patrick/bin/code, and added it to my path so when I type codeecho $PATH/code it comes up correctly like so:   precode/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/User/patrick/bin:/usr/local/go/bin`.  /code/pre  I already added the line code#!/usr/bin/env python/code to the top of the file, I already did codechmod 700 testscript/code, but I still can't call codetestscript/code from anywhere. Have things changed since OSX Yosemite?            <answer169116> Make sure the script is executable. code chmode +x filename /code  Make sure the script has the appropriate shebang line at the top: code #!/usr/bin/env python /code  Double check the path, it should be code/Users/code (note the strongs/strong at the end)  <comment198521> Case may matter, is it **P**atrick or **p**atrick? Also, what error message are you getting? Does running it via `~/bin/testscript` work? <comment198524> No @patrix, OS X is case-insensitive. <comment198529> Ah, probably, yes. But it's `/User**s**`, at least here. <comment198554> @partix Yeah, thats right, certain UNIX tools are case sensitive. <comment198568> @patrix Wow, that was it... I just noticed that on OSX it says "Users" instead of "User"... It works now, thanks.